### PR TITLE
EL-1736: Set up Github Action Pipeline 

### PR DESCRIPTION
**Pull request by MazOneTwoOne**
- Set up Github Action Pipeline based on skeleton
- The deployment to UAT is dependant on the helm ticket to be merged in
- Removed the `trivy` scan image job as this is a feature that is work in progress (as discussed with the skeleton owner)
- Reverted `starlette` PR changes as this was preventing the test job to succeed
  - if we want that we have to [follow these notes](https://fastapi.tiangolo.com/release-notes/#01120)
- have set up the `ECR_REGISTRY` secret for this project

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 sonar.organization=ministryofjustice
-sonar.projectKey=add-your-project-key-here
+sonar.projectKey=ministryofjustice_laa-pola
```
Comment (willc-work) [inline sonar-project.properties:2]: really minor, can we use hyphens or underscores but not mix them if possible

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+          KUBE_NAMESPACE: ${{ secrets.KUBE_NAMESPACE }}
+          KUBE_CLUSTER: ${{ secrets.KUBE_CLUSTER }}
+        run: |
+          echo "${{ secrets.KUBE_CERT }}" > ca.crt
+          kubectl config set-cluster ${KUBE_CLUSTER} --certificate-authority=./ca.crt --server=https://${KUBE_CLUSTER}
+          kubectl config set-credentials deploy-user --token=${{ secrets.KUBE_TOKEN }}
+          kubectl config set-context ${KUBE_CLUSTER} --cluster=${KUBE_CLUSTER} --user=deploy-user --namespace=${KUBE_NAMESPACE}
+          kubectl config use-context ${KUBE_CLUSTER}
+      
+      - name: Fetch and set SHARED_IP_RANGES_LAA environment variable
```
Comment (willc-work) [inline .github/workflows/deploy.yml:52]: what do we need this for?Will POLA only be accessible from VPN?

**Review by willc-work — APPROVED** [review]
1 question and 1 requested change. neither are blockers to merging.

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (MazOneTwoOne) [reply .github/workflows/deploy.yml:52]: @willc-work I think that's the intention with having this in the skeleton, the APIs that people build may not always be public. 

I've kept in as not sure how public POLA will be.

File: sonar-project.properties (hunk 1 above)
Comment (MazOneTwoOne) [reply sonar-project.properties:2]: That is a fair point. Annoyingly, I can't see a way to edit this or delete the project I created.   

Screenshot 2024-10-30 at 09 16 41

Comment (willc-work) [reply sonar-project.properties:2]: no worries

**Pull request merged**